hs_file_types/models/reftimeseries.py

```diff
@@ -0,0 +1,922 @@
+                            td(self.return_type)
+                        with tr():
+                            get_th('Reference Type')
+                            td(self.reference_type)
+
+        return root_div.render(pretty=True)
+
+
+class RefTimeseriesFileMetaData(AbstractFileMetaData):
+    # the metadata element models are from the hs_core app
+    model_app_label = 'hs_core'
+    # field to store the content of the json file (the file that is part
+    # of the RefTimeseriesLogicalFile type
+    json_file_content = models.TextField()
+
+    @property
+    def title(self):
+        """get the title associated with this ref time series"""
+        json_data_dict = self._json_to_dict()
+        if 'title' in json_data_dict['timeSeriesReferenceFile']:
+            return json_data_dict['timeSeriesReferenceFile']['title']
+        return ''
+
+    @property
+    def abstract(self):
+        """get the abstract associated with this ref time series"""
+        json_data_dict = self._json_to_dict()
+        if 'abstract' in json_data_dict['timeSeriesReferenceFile']:
+            return json_data_dict['timeSeriesReferenceFile']['abstract']
+        return ''
+
+    @property
+    def file_version(self):
+        """get the file version associated with this ref time series"""
+        json_data_dict = self._json_to_dict()
+        if 'fileVersion' in json_data_dict['timeSeriesReferenceFile']:
+            return json_data_dict['timeSeriesReferenceFile']['fileVersion']
+        return ''
+
+    @property
+    def key_words(self):
+        """get a list of all keywords associated with this ref time series"""
+        # had to name this property as key_words since the parent class has a model field keywords
+        json_data_dict = self._json_to_dict()
+        if 'keyWords' in json_data_dict['timeSeriesReferenceFile']:
+            return json_data_dict['timeSeriesReferenceFile']['keyWords']
+        return []
+
+    @property
+    def sample_mediums(self):
+        """get a list of all sample mediums associated with this ref time series"""
+        sample_mediums = []
+        for series in self.serieses:
+            if "sampleMedium" in series:
+                if series['sampleMedium'] not in sample_mediums:
+                    sample_mediums.append(series['sampleMedium'])
+        return sample_mediums
+
+    @property
+    def value_counts(self):
+        """get a list of all value counts associated with this ref time series"""
+        value_counts = []
+        for series in self.serieses:
+            if "valueCount" in series:
+                if series['valueCount'] not in value_counts:
+                    value_counts.append(series['valueCount'])
+        return value_counts
+
+    @property
+    def serieses(self):
+        json_data_dict = self._json_to_dict()
+        return json_data_dict['timeSeriesReferenceFile']['referencedTimeSeries']
+
+    @property
+    def time_serieses(self):
```

Comment:
@pkdash Perhaps better to call this property ```time_series_list``` rather than ```time_serieses``` since I don't think ```serieses``` is a valid plural word for ```series```